File: backend/app/schemas/service_discovery.py
from pydantic import BaseModel


class ServiceDiscoveryNamespaceOption(BaseModel):
    label: str
    namespace: str


class ServiceDiscoveryConfigResponse(BaseModel):
    configured: bool
    namespace_options: list[ServiceDiscoveryNamespaceOption]


class ServiceDiscoveryIngressItem(BaseModel):
    name: str
    namespace: str
    hosts: list[str]
    ingress_class_name: str | None = None
    load_balancers: list[str] = []
    tls_hosts: list[str] = []


class ServiceDiscoveryIngressListResponse(BaseModel):
    namespace: str
    items: list[ServiceDiscoveryIngressItem]

File: frontend/src/api/serviceDiscovery.ts
import { cachedGet, invalidateCache } from './request'

export interface ServiceDiscoveryNamespaceOption {
  label: string
  namespace: string
}

export interface ServiceDiscoveryConfigResponse {
  configured: boolean
  namespace_options: ServiceDiscoveryNamespaceOption[]
}

export interface ServiceDiscoveryIngressItem {
  name: string
  namespace: string
  hosts: string[]
  ingress_class_name?: string | null
  load_balancers: string[]
  tls_hosts: string[]
}

export interface ServiceDiscoveryIngressListResponse {
  namespace: string
  items: ServiceDiscoveryIngressItem[]
}

export function getServiceDiscoveryConfig(force = false) {
  return cachedGet<ServiceDiscoveryConfigResponse>('/service-discovery/config', { force, ttl: 300_000 })
}

export function getServiceIngresses(namespace?: string, force = false) {
  return cachedGet<ServiceDiscoveryIngressListResponse>('/service-discovery/ingresses', {
    params: namespace ? { namespace } : {},
    force,
    ttl: 60_000,
  })
}

export function invalidateServiceDiscoveryCache() {
  invalidateCache('/service-discovery')
}
